app/components/ui/status-button.tsx

```diff
@@ -25,17 +25,26 @@ export const StatusButton = React.forwardRef<
 	})
 	const companion = {
 		pending: delayedPending ? (
-			<div className="inline-flex h-6 w-6 items-center justify-center">
+			<div
+				role="status"
```

Comment:
Good idea on this. But we need to give these an accessible name so screen readers can announce what the current status is. I think we should probably add a `title` prop to `Icon` and if it's specified, then we add a `<title>` to the `<svg>` that's rendered (next to the `<use>`).

As an additional benefit, way you can select it by it's accessible name as well as its role.

Reply:
How beautiful is that, the browser will also show a tooltip on hover